packages/application/src/frontend.ts

```diff
@@ -203,6 +203,72 @@ export abstract class JupyterFrontEnd<
     }
   }
 
+  /**
+   * Override keydown handling to prevent command shortcuts from preventing user input.
+   *
+   * This introduces a slight delay to the command invocation, but no delay to user input.
+   */
+  protected evtKeydown(event: KeyboardEvent): void {
+    // Process select keys which may call `preventDefault()` immediately
+    if (
+      ['Tab', 'ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(
+        event.key
+      )
+    ) {
```

Comment:
These keys were selected from manual testing. Maybe we could also check for modifiers not being present. There are more keys to consider (Home/End, Page Down/Up - anything else?) that may have an action which needs to be prevented immediately, e.g. in the context of windowed notebook.